### DISCUSSION TITLE
`req` / `user` access in condition callback

### DISCUSSION THREAD

**Discussion by ssyberg** [payloadcms/payload#2541] [Feature Requests & Ideas]
We have many uses cases for field level access control were using `access` is a bit clunky and it would be much nicer/easier to just hide fields that users shouldn't touch. Currently condition callbacks only accept `data` and `siblingData`

**Comment by DanRibbens** [comment]
I looked at adding `user` to the arguments for condition functions and I see that the type on this function isn't ideal for expanding it. Since it is called with `(data, siblingData)` I have to add user as the 3rd argument on the function. This works, but I am fairly confident there will be a breaking change as we're going to want move to an argument object to make this future proof.

I'm moving ahead with (data, siblingData, { user }) for now.

**Reply by DanRibbens** [reply]
https://github.com/payloadcms/payload/pull/2543

**Reply by ssyberg** [reply]
Yea I was a bit surprised by the callback args in `condition` it does seem to break the pattern across the other hooks/callbacks but this solution seems reasonable to me!

**Reply by DanRibbens** [reply]
We can break from this in 2.0.
